### ISSUE TITLE
Parse empty list of type arguments

### ISSUE THREAD

**Issue by phischu** [effekt-lang/effekt#714]
The following program does not parse with the error "expected identifier"

```
def f() = 42

def main(): Unit = {
  println(f[]())
}
```

I would like to be able to write an empty list of type arguments to disambiguate overloads. Indeed, writing no type argument list should mean "inferred", which is different from "annotated as empty list".

On the definition site this is not necessary, since omitting the list of type parameters means "empty list" unambiguously.

**Comment by jiribenes** [comment]
I agree with the underlying premise: it would be nice to be able to specify "I want the version with no type params" to the overload resolution.

**Comment by b-studios** [comment]
Discussed: not controversial, just ugly.

**Comment by b-studios** [comment]
Once there is a good use of this we can consider :)

**Comment by jiribenes** [comment]
We're not planning to address the issue by parsing `foo[]` in the near future; dunno why it got reopened.
If you disagree, feel free to reopen. :)
